src/player.h

```diff
@@ -1452,6 +1460,20 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         const pathfinding_settings &get_pathfinding_settings() const override;
         std::set<tripoint> get_path_avoid() const override;
 
+        /**
+         * Try to disarm the NPC. May result in fail attempt, you receiving the wepon and instantly wielding it,
+         * or the weapon falling down on the floor nearby. NPC is always getting angry with you.
+         * @param target Target NPC to disarm
+         */
+        void disarm( npc &target );
```

Comment:
That was my idea from the beginning to make `steal` and `disarm` attacks usable by npcs and agains zombies.
But it seems non-trivial for now, since `player` lacks npc-like methods like `make_angry`. Dynamic casts would help, but might look ugly.